### PR TITLE
Add role back to the ServerSet Status

### PR DESCRIPTION
**Pull request by fghimpeteanu-ionos**
### Description of your changes

This PR adds:

## Checklist

I have:

- [ ] Add PR name as appropriate (e.g. `feat`/`fix`/`doc`/`test`/`refactor`)
- [ ] Run `make reviewable` and `make crds.clean` to ensure the PR is ready for review
- [ ] Add or update tests (if applicable)
- [ ] Add or update Documentation using `make docs.update` (if applicable)
- [ ] Update `docs/CHANGELOG.md` file (label: `upcoming release`)
- [ ] Check Sonar Cloud Scan
- [ ] Update Github or Jira Issue

### CODE REVIEW

File: apis/compute/v1alpha1/serverset_types.go [hunk 1]
```diff
@@ -118,6 +127,8 @@ type ServerSetObservation struct {
 
 // ServerSetReplicaStatus are the observable fields of a ServerSetReplicaStatus.
 type ServerSetReplicaStatus struct {
+	// +kubebuilder:validation:Enum=ACTIVE;PASSIVE
+	Role string `json:"role"`
```
Comment (ionos-jk) [inline apis/compute/v1alpha1/serverset_types.go:131]: You can use Role Role here

File: internal/clients/ionoscloud_test.go [hunk 2]
```diff
@@ -15,23 +13,23 @@ import (
...
-func setComputeDefaults(cfg *ionos.Configuration) {
-	cfg.HTTPClient = http.DefaultClient
-	cfg.UserAgent = fmt.Sprintf("%v/%v_ionos-cloud-sdk-go/v%v", UserAgent, version.Version, ionos.Version)
-}
-
-func setDbaaSDefaults(cfg *ionosdbaas.Configuration) {
-	cfg.HTTPClient = http.DefaultClient
-	cfg.UserAgent = fmt.Sprintf("%v/%v_ionos-cloud-sdk-go-dbaas-postgres/v%v", UserAgent, version.Version, ionosdbaas.Version)
-}
+// TODO: Fix this tests in https://github.com/ionos-cloud/crossplane-provider-ionoscloud/issues/204
```
Comment (ionos-jk) [inline internal/clients/ionoscloud_test.go:18]: why are those tests failing?

File: internal/controller/serverset/serverset.go [hunk 3]
```diff
@@ -436,6 +457,10 @@ func (e *external) Delete(ctx context.Context, mg resource.Managed) error {
 
 // AreServersUpToDate checks if replicas and template params are equal to server obj params
 func AreServersUpToDate(templateParams v1alpha1.ServerSetTemplateSpec, servers []v1alpha1.Server) bool {
```
Comment (ionos-jk) [inline internal/controller/serverset/serverset.go:459]: keep in mind when `replicas: 0` is set, the servers would be up2date

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -62,6 +62,38 @@ jobs:
...
+        with:
+          path: ${{ steps.go.outputs.cache }}
+          key: ${{ runner.os }}-build-lint-${{ hashFiles('**/go.sum') }}
+          restore-keys: ${{ runner.os }}-build-lint-
+      - name: Cache Go Dependencies
+        uses: actions/cache@v2
+        with:
+          path: .work/pkg
+          key: ${{ runner.os }}-pkg-${{ hashFiles('**/go.sum') }}
+          restore-keys: ${{ runner.os }}-pkg-
```
Comment (ionos-jk) [inline .github/workflows/ci.yml:92]: setup-go@v4 has caching enabled by default

https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-go#caching-dependencies
https://github.com/actions/setup-go

**Review by ionos-jk — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 4 above)
Comment (fghimpeteanu-ionos) [reply .github/workflows/ci.yml]: Thank you for the tip :pray:

File: apis/compute/v1alpha1/serverset_types.go (hunk 1 above)
Comment (fghimpeteanu-ionos) [reply apis/compute/v1alpha1/serverset_types.go]: Done

**Review by cbeti-ionos — APPROVED** [review]

File: internal/clients/ionoscloud_test.go (hunk 2 above)
Comment (cbeti-ionos) [reply internal/clients/ionoscloud_test.go]: Logger object in `ionoscloud.Configuration` is not comparable through DeepEqual

**Review by ionos-jk — APPROVED** [review]

**Pull request merged**